docs/site/src/components/EffortBox/index.tsx

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// This component is used as part of the effort plugin.
+// It appears at the top of any guide with an `effort`
+// rating (small, medium, large) in its frontmatter.
+
+import React from "react";
+import Admonition from "@theme/Admonition";
+
+export default function EffortBox(props) {
+  if (!props.effort) {
+    return;
+  }
+  const effort =
+    props.effort[0] === "s"
+      ? "basic"
+      : props.effort[0] === "m"
+        ? "involved"
+        : "advanced";
+  const time =
+    effort === "basic"
+      ? "30-45 minutes"
+      : effort === "involved"
+        ? "1-1.5 hours"
+        : "2 hours or more";
```

Comment:
I find in JS/TS that for this kind of pattern, it's neater to split things out into its own function:

```ts
function timeAndEffort(effort: string): [string, string] {
  if (effort[0] == "s") {
    return ["30-45 minutes", "basic"];
  } else if (effort[0] == "m") {
    return ["1-1.5 hours", "involved"];
  } else {
    return ["2 hours or more", "advanced"];
  }
}

const [time, effort] = timeAndEffort(props.effort);
```